apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/RetryTest.cs

```diff
@@ -30,23 +29,30 @@ namespace Google.Cloud.Storage.V1.Tests;
 /// </summary>
 public class RetryTest
 {
-    /// <summary>
-    /// A set of retry options using default timing, but a predicate that retries on 404 (NotFound) only.
-    /// </summary>
-    private static readonly RetryOptions s_customPredicateOptions = new RetryOptions(RetryTiming.Default, RetryPredicate.FromErrorCodes(404));
+    private static readonly RetryTiming s_customRetryTiming = new(TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(5), 2);
+    private static readonly RetryPredicate s_customPredicate = RetryPredicate.FromErrorCodes(404, 400);
+    private static readonly RetryOptions s_customRetryOptions = new(s_customRetryTiming, s_customPredicate);
 
     public static TheoryData<TestCase> TestCases { get; } = new TheoryData<TestCase>
     {
-        new TestCase("SimpleSuccess", RetryOptions.Never, new[] { OK }, new[] { 0 }, true),
-        new TestCase("ImmediateFailure", RetryOptions.IdempotentRetryOptions, new[] { BadRequest }, new[] { 0 }, false),
-        new TestCase("SingleRetry", RetryOptions.IdempotentRetryOptions, new[] { InternalServerError, OK }, new[] { 0, 1 }, true),
-        new TestCase("DoubleRetry", RetryOptions.IdempotentRetryOptions, new[] { InternalServerError, BadGateway, OK }, new[] { 0, 1, 3 }, true),
-        // We default to a maximum of 3 requests
+        new TestCase("SimpleSuccess_NoRetry", RetryOptions.IdempotentRetryOptions, new[] { OK }, new[] { 0 }, true),
+        new TestCase("SuccessAfterMaxRetry", RetryOptions.IdempotentRetryOptions, new[] { BadGateway, BadGateway, OK }, new[] { 0, 1, 3 }, true),
+        new TestCase("FailureAfterMaxRetry", RetryOptions.IdempotentRetryOptions, new[] { BadGateway, BadGateway, BadGateway }, new[] { 0, 1, 3 }, false),
+
+        new TestCase("CustomTiming_SuccessAfterRetry",  s_customRetryOptions, new[] { NotFound, BadRequest, OK }, new[] { 0, 1, 3 }, true),
```

Comment:
I'd remove the CustomTiming_ and CustomPredicate_ part of the name of all of these tests, given that they're all using the same options.

But I *would* create another test with *different* timing and test that. At the moment, RetryHandler could use hard-coded timing, ignoring the timing in the options, and the tests would still pass.

Reply:
(If you just change the custom retry options to use an exponential factor of 3 instead of 2, but keep the tests with IdempotentRetryOptions, that would make sure we cover timing.)

Reply:
Changed the timing in s_customRetryOptions itself , so timing can also be tested and cause failure in case of mismatch